Yosemite/YosemiteTests/Stores/OrderStoreTests.swift

```diff
@@ -563,6 +563,163 @@ final class OrderStoreTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    func test_optimistic_update_order_customer_note_correctly() {
+        // Given
+        let orderStore = OrderStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+
+        let originalOrder = sampleOrder()
+        let updatedOrder = originalOrder.copy(customerNote: "Updated!")
+
+        orderStore.upsertStoredOrder(readOnlyOrder: originalOrder, in: viewStorage)
+
+        /// Update: Expected Customer note is actually coming from `order.json` (Customer note == "")
+        network.simulateResponse(requestUrlSuffix: "orders/963", filename: "order")
+
+        // When
+        let result: Result<Networking.Order, Error> = waitFor { promise in
+            let action = OrderAction.updateOrderOptimistically(siteID: self.sampleSiteID, order: updatedOrder, fields: [.customerNote]) { result in
+                promise(result)
+            }
+            orderStore.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        let storageOrder = storageManager.viewStorage.loadOrder(siteID: sampleSiteID, orderID: sampleOrderID)
+       XCTAssertEqual(storageOrder?.customerNote, updatedOrder.customerNote)
```

Comment:
Super nit: I think my suggestion earlier included the removal of a space 😅

Reply:
Good catch 🕵🏼‍♂️ 🙂